src/main/org/firebirdsql/jdbc/AbstractResultSet.java

```diff
@@ -277,7 +282,15 @@ public void setFieldData(byte[] data) {
                 }
             };
 
-            fields[i] = FBField.createField(rowDescriptor.getFieldDescriptor(i), dataProvider, gdsHelper, cached);
+            final FBField field = FBField.createField(rowDescriptor.getFieldDescriptor(i), dataProvider, gdsHelper, cached);
+
+            if (field.isNeedClose()) {
+                if (closeableFields == null)
+                    closeableFields = new LinkedList();
```

Comment:
Please use an `ArrayList` instead: linked list is almost always the wrong choice: it uses a lot of space, and is almost always slower than an `ArrayList`.